docs/website/contents/for-developers/utxo-hd/utxo-hd.md

```diff
@@ -0,0 +1,436 @@
+
+## Why does Cardano need UTXO-HD
+
+Cardano is built following the UTXO model. This means that the Ledger state
+contains a map from _transaction inputs_ to _transaction outputs_. A transaction
+might consume some of those entries and produce new ones. Each entry is owned by
+an account which is the one that can spend it.
+
+The UTXO set is an always growing data structure. Currently the `cardano-node`
+runs in about 16GBs of RAM but this amount will keep growing as more traffic
+takes place on the network. This is bad for decentralization and sustainability of
+the network as eventually only powerful machines would be able to participate on
+it.
+
+To improve decentralization, a decision was made to move this data to persistent
+storage which, albeit slower, is much cheaper than RAM. The Consensus layer is
+reworked so that the flow of data now allows for UTXO entries to come from some
+backend storage, which might be on disk or in memory trading memory for speed.
+
+The UTXO-HD feature provides two backends which can be chosen in cardano-node's
+configuration file:
+
+- `LedgerDBBackend: V2InMemory`
+- `LedgerDBBackend: V1LMDB`
+
+How these backends work is shown below in this document.
+
+## UTXO-HD design in Consensus
+
+> ℹ️ We are going to focus on Shelley based eras, ignoring Byron for now.
+
+### The `NewEpochState` data structure
+The Ledger layer defines the data structure that holds the state of the blockchain
+after applying some number of blocks, the `NewEpochState`. Among other things,
+this data structure holds a UTXO set which is a `Map` from
+`TxIn (EraCrypto era)` to `TxOut era`.
+
+In order to apply the different Ledger operations, there is no need for this set
+to be complete at all times, as only entries consumed by the transactions will be
+accessed. When given a block or a transaction, the Ledger code provides functions
+for getting the set of keys that would be necessary to exist in the UTXO set for
+that block or transaction to apply correctly.Taking advantage of this, the Consensus layer will modify this
+container such that it only contains the entries necessary for the Ledger rules. 
+
+### Shelley instantiation and ledger tables
+
+The `LedgerState (ShelleyBlock proto era)` data family instances are augmented
+with a new field which will hold these entries that will be extracted from and
+injected to the `NewEpochState` before calling the Ledger rules. This new field
+(which we call _ledger tables_) is a container-like structure parametrized by a
+`Key` and `Value` type families.
+
+```diff haskell
+data instance LedgerState (ShelleyBlock proto era) mk = ShelleyLedgerState {
+       shelleyLedgerTip        :: !(WithOrigin (ShelleyTip proto era))
+     , shelleyLedgerState      :: !(SL.NewEpochState era)
+     , shelleyLedgerTransition :: !ShelleyTransition
++    , shelleyLedgerTables     :: !(LedgerTables (LedgerState (ShelleyBlock proto era)) mk)
+    }
```

Comment:
I think it's important to show that the *consensus* ledger state has to include tables (or it can ignore them, but at least it has to have an mk parameter), but a tutorial could probably explain this as well